src/JamesHalsall/Licenser/Licenser.php

```diff
@@ -210,4 +194,83 @@ private function log($message)
 
         $this->output->writeln($message);
     }
+
+    /**
+     * @param string  $path The path to the files/directory
+     */
+    public function check($path)
+    {
+        foreach ( $this->getFiles($path) as $file) {
+            $this->checkFile($file);
+        }
+    }
+
+    /**
+     * Checks the header of 1 file
+     * @param SplFileInfo $file
+     */
+    private function checkFile(SplFileInfo $file)
+    {
+        $tokens = token_get_all($file->getContents());
+        $licenseTokenIndex = $this->getLicensetokenIndex($tokens);
+
+        if ($licenseTokenIndex === null) {
+            $this->log(sprintf('Missing license header in "%s"', $file->getRealPath()));
+        } elseif ($tokens[$licenseTokenIndex][1] != $this->getLicenseAsComment()) {
+            $this->log(sprintf('Different license header in "%s"', $file->getRealPath()));
+        }
+    }
+
+    /**
+     * @param $path
+     * @return SplFileInfo[]
+     */
+    private function getFiles($path)
+    {
+        if (is_file($path)) {
+            return [new SplFileInfo($path, '', '')];
+        } else {
+            $iterator = $this->finder->name('*.php')->in(realpath($path));
+        }
+        return $iterator;
```

Comment:
this block can be simplified, instead of the `else` part of the condition we can just do `return $this->finder->name(...`